src/main/java/uk/nhs/digital/nhsconnect/nhais/service/RecepProducerService.java

```diff
@@ -36,96 +41,102 @@ private String toEdifact(List<Segment> segments) {
-//        return new MessageTrailer(5 + refSegmentCount);
-//    }
-//
-//    private DateTimePeriod mapToRecepTranslationDateTime(Interchange receivedInterchangeFromHa) {
-//        return new DateTimePeriod(
-//            receivedInterchangeFromHa.getTranslationDateTime().getTimestamp(),
-//            receivedInterchangeFromHa.getTranslationDateTime().getTypeAndFormat());
-//    }
-//
-//    private RecepHeader mapToRecipInterchangeHeader(Interchange interchange) {
-//        var recepSender = interchange.getInterchangeHeader().getRecipient();
-//        var recepRecipient = interchange.getInterchangeHeader().getSender();
-//
-//        return new RecepHeader(recepSender, recepRecipient, interchange.getInterchangeHeader().getTranslationTime());
-//    }
-//
-//    private RecepMessageHeader mapToRecipMessageHeader(Interchange interchange) {
-//        return new RecepMessageHeader()
-//            .setSequenceNumber(interchange.getMessageHeader().getSequenceNumber());
-//    }
-//
-//    private RecepBeginningOfMessage mapToRecepBeginningOfMessage(Interchange interchange) {
-//        return new RecepBeginningOfMessage(interchange.getTranslationDateTime().getTimestamp());
-//    }
-//
-//    private RecepNameAndAddress mapToRecepNameAndAddress(Interchange interchange) {
-//        return new RecepNameAndAddress(interchange.getInterchangeHeader().getSender());
-//    }
-//
-//    private ReferenceMessageRecep mapToReferenceMessageRecep(Interchange interchange) {
-//        return new ReferenceMessageRecep(
-//            interchange.getMessageHeader().getSequenceNumber(), ReferenceMessageRecep.RecepCode.SUCCESS);
-//    }
-//
-//    private ReferenceInterchangeRecep mapToReferenceInterchangeRecep(Interchange interchange) {
-//        return new ReferenceInterchangeRecep(
-//            interchange.getInterchangeHeader().getSequenceNumber(),
-//            ReferenceInterchangeRecep.RecepCode.RECEIVED,
-//            interchange.getInterchangeTrailer().getNumberOfMessages());
-//    }
+    private List<Segment> mapEdifactToRecep(Interchange receivedInterchangeFromHa) throws EdifactValidationException {
+        List<Segment> recepMessageSegments = new ArrayList<>();
+
+        var recepInterchangeHeader = mapToRecipInterchangeHeader(receivedInterchangeFromHa);
+        var recepMessageHeader = mapToRecipMessageHeader(receivedInterchangeFromHa);
+
+        var referenceMessageRecepSegments = receivedInterchangeFromHa.getMessages().stream()
+            .map(this::mapToReferenceMessageRecep)
+            .collect(Collectors.toList());
+
+        recepMessageSegments.add(recepInterchangeHeader);
+        recepMessageSegments.add(recepMessageHeader);
+        var recepBeginningOfMessage = mapToRecepBeginningOfMessage(receivedInterchangeFromHa);
+        recepMessageSegments.add(recepBeginningOfMessage);
+        recepMessageSegments.add(mapToRecepNameAndAddress(receivedInterchangeFromHa));
```

Comment:
This segment is being translated incorrectly but I have raised a bug in the next sprint to address it.